dev/TeachingTip/TeachingTip.cpp

```diff
@@ -867,6 +952,29 @@ void TeachingTip::OnActionButtonClicked(const winrt::IInspectable&, const winrt:
 void TeachingTip::OnPopupOpened(const winrt::IInspectable&, const winrt::IInspectable&)
 {
     StartExpandToOpen();
+
+    if (auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*this))
+    {
+        if (auto teachingTipPeer = peer.as<winrt::TeachingTipAutomationPeer>())
+        {
+            winrt::hstring notificationString;
+            auto appName = winrt::ApplicationModel::Package::Current().DisplayName();
```

Comment:
[Avoid declaring locals without initialization](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es22-dont-declare-a-variable-until-you-have-a-value-to-initialize-it-with). In cases like this where you don't know which branch will produce the value then I like using lambdas. It also facilitates refactoring to helper functions.

auto notificationString = [this]() {
            auto appName = winrt::ApplicationModel::Package::Current().DisplayName();
            if (appName.size() > 0)
            {
                return StringUtil::FormatString(
                    ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipNotification),
                    appName.data(),
                    winrt::AutomationProperties::GetName(m_popup.get()).data());
            }
            else
            {
                return StringUtil::FormatString(
                    ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipNotificationWithoutAppName),
                    winrt::AutomationProperties::GetName(m_popup.get()).data());
            }
};